plugins/woocommerce/tests/e2e-pw/bin/woocommerce-cleanup.php

```diff
@@ -0,0 +1,369 @@
+
+	$files = array_diff( scandir( $dir ), array( '.', '..' ) );
+
+	foreach ( $files as $file ) {
+		$path = $dir . DIRECTORY_SEPARATOR . $file;
+
+		if ( is_dir( $path ) ) {
+			wc_cleanup_directory( $path, $excluded_file_names );
+		} else {
+			$file_name = pathinfo( $file, PATHINFO_FILENAME );
+			if ( ! in_array( $file_name, $excluded_file_names, true ) ) {
+				wp_delete_file( $path );
+			}
+		}
+	}
+
+	// Remove the empty directory if it's not the base upload directory.
+	if ( wp_upload_dir()['basedir'] !== $dir ) {
+		if ( function_exists( 'wp_delete_directory' ) ) {
+			wp_delete_directory( $dir );
+		} elseif ( function_exists( 'WP_Filesystem' ) ) {
+			WP_Filesystem();
+			global $wp_filesystem;
+			$wp_filesystem->rmdir( $dir );
+		} else {
+			// Fallback for WordPress versions that don't have wp_delete_directory or WP_Filesystem.
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir
+			rmdir( $dir );
+		}
+	}
+}
+
+/**
+ * Remove all WooCommerce taxes and tax classes.
+ */
+function wc_cleanup_taxes() {
+	global $wpdb;
+
+	// Remove all tax rates.
+	$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}woocommerce_tax_rates" );
+	$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}woocommerce_tax_rate_locations" );
+
+	// Remove all tax classes except the default ones.
+	$wpdb->query( "DELETE FROM {$wpdb->prefix}woocommerce_tax_classes WHERE slug NOT IN ('standard', 'reduced-rate', 'zero-rate')" );
+
+	// Reset tax options.
+	update_option( 'woocommerce_tax_classes', 'Standard\nReduced rate\nZero rate' );
+	update_option( 'woocommerce_tax_display_shop', 'excl' );
+	update_option( 'woocommerce_tax_display_cart', 'excl' );
+	update_option( 'woocommerce_tax_total_display', 'itemized' );
+
+	// Disable taxes.
+	update_option( 'woocommerce_calc_taxes', 'no' );
+
+	// Clear the tax transients.
+	WC_Cache_Helper::invalidate_cache_group( 'taxes' );
+}
+
+/**
+ * Truncate a WooCommerce analytics table.
+ *
+ * @param string $table_name The name of the table to truncate, without the prefix.
+ */
+function wc_cleanup_analytics_table( $table_name ) {
+	global $wpdb;
+	$full_table_name = $wpdb->prefix . $table_name;
+
+	// Check if the table exists before attempting to truncate.
+	$table_exists = $wpdb->get_var(
+		$wpdb->prepare(
+			'SHOW TABLES LIKE %s',
+			$full_table_name
+		)
+	) === $full_table_name;
+
+	if ( $table_exists ) {
+		$wpdb->query( $wpdb->prepare( 'TRUNCATE TABLE %i', $full_table_name ) );
+	} elseif ( defined( 'WP_DEBUG_LOG' ) && WP_DEBUG_LOG ) {
+		// Use WP_DEBUG_LOG instead of error_log.
+		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		error_log( "Table $full_table_name does not exist." );
+	}
+}
+
+/**
+ * Reset the WooCommerce site.
+ */
+function wc_cleanup_reset_site() {
+
+	// Remove all coupons.
+	$coupons = get_posts(
+		array(
+			'post_type'   => 'shop_coupon',
+			'numberposts' => -1,
+		)
+	);
+	foreach ( $coupons as $coupon ) {
+		wp_delete_post( $coupon->ID, true );
```

Comment:
@lanej0, I noticed coupons are left in the trash, and some tests (at least one on the WPCOM site) are failing. 

Here is an example:
![image](https://github.com/user-attachments/assets/7ec3124b-e02b-49f6-a16c-927a3744e985)

Because of this, the "Create your first coupon" button is missing, and the test is failing. 

Is it possible to completely remove coupons? I thought `true` parameter should do that, but not in this case.

Reply:
I just pushed an update and updated both Pressable and WPCom with the latest. It seems to be deleting them now, can you confirm?

Reply:
Have you installed it on Pressable and WPCOM? The one in the trash was still in the trash after the cleanup

Reply:
I've added this new part of the code in the plugin, and it emptied the trash as well:

```
	// Remove all coupons that are in the trash.
	$trash_coupons = get_posts(
	    array(
	        'post_type'   => 'shop_coupon',
	        'post_status' => 'trash',
	        'numberposts' => -1,
	    )
	);

	foreach ( $trash_coupons as $coupon ) {
	    wp_delete_post( $coupon->ID, true );
	}
```